components/ui/PortalPopup.tsx

```diff
@@ -0,0 +1,228 @@
+/*
+ * Sonar UI Common
+ * Copyright (C) 2019-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import { throttle } from 'lodash';
+import * as React from 'react';
+import { createPortal, findDOMNode } from 'react-dom';
+import ScreenPositionFixer from '../controls/ScreenPositionFixer';
+import { Popup, PopupPlacement } from './popups';
+
+interface PopupProps {
+  arrowOffset?: number;
+  children: React.ReactNode;
+  className?: string;
+  noPadding?: boolean;
+  overlay: React.ReactElement<PopupProps> | undefined;
+  placement?: PopupPlacement;
+}
+
+interface Measurements {
+  height: number;
+  left: number;
+  top: number;
+  width: number;
+}
+
+type State = Partial<Measurements>;
+
+function isMeasured(state: State): state is Measurements {
+  return state.height !== undefined;
+}
+
+export default class PortalPopup extends React.Component<PopupProps, State> {
```

Comment:
Not sure if that is useful but you can remove `children` from the PopupProps and use `react.PropsWithChildren<Props>` here.

Reply:
Yes, I could but I don't want the children to be optional like in `React.PropsWithChildren`, so I just thought it was more straightforward to define it myself...